src/tasks/copy-template-files.ts

```diff
@@ -321,11 +344,27 @@ export async function copyTemplateFiles(options: Options, templateDir: string, t
       await setUpExternalExtensionFiles(options, tmpDir);
     }
 
+    if (options.solidityFramework) {
+      const externalExtensionSolidityPath = path.join(
+        externalExtensionPath,
+        "packages",
+        options.solidityFramework,
+        "contracts",
+      );
+      // if external extension does not have solidity framework, we copy the example contracts
+      if (!fs.existsSync(externalExtensionSolidityPath) && exampleContractsPath) {
+        await copyExtensionFiles(options, exampleContractsPath, targetDir);
+      }
+    }
+
     await copyExtensionFiles(options, externalExtensionPath, targetDir);
   }
 
+  if (!options.externalExtension && exampleContractsPath) {
```

Comment:
Just leaving a note here. No need to change anything, but since this is the most tricky file in the who repo... I think it's worth it to talk about the little things.

The `exampleContractsPath` variable is confusing (same pattern as `solidityFrameworkPath`), since the variable can be null (when no solidity framework is selected). In my mind, the path always exists (it's part of the repo) but sometimes we don't want to copy it over (when no solidityFramework).

So when I see this condition `if (!options.externalExtension && exampleContractsPath) {` I have to double-check. 

`if (!options.externalExtension && options.solidityFramework) {` reads better to me.

Little think I know haha.  Sorry!

Reply:
Ohhh no lol trust me I actually spend time thinking about same issue and condition and wanted to have the same condition but typescript is not being smart : 

![Screenshot 2024-10-22 at 2 17 20 PM](https://github.com/user-attachments/assets/72c6e4a2-f631-4606-b84a-739d9d537df7)

We know that `exampleContractsPath` will not be null if `options.solidityFramework` is present and when we do 

`if (!options.externalExtension && options.solidityFramework)` 

It should have known that `exampleContractsPath` will not be null and won't complain while calling `copyExtensionFiles`. 

I actually thought of maybe creating `exampleContractsPath` inside the condition but it was also being used at above and below places. 

for now I updated the condition : 

`if (!options.externalExtension && options.solidityFramework && exampleContractsPath)`

Reply:
hahahah Thanks! <3